pyproject.toml

```diff
@@ -21,3 +21,26 @@ before-test = "pip install --only-binary=numpy,scipy numpy scipy"
 [tool.cibuildwheel.linux]
 before-all = "yum install -y wget && {package}/tools/install_rust.sh"
 environment = 'PATH="$PATH:$HOME/.cargo/bin" CARGO_NET_GIT_FETCH_WITH_CLI="true"'
+
+
+[tool.ruff]
+line-length=105
+
+[tool.ruff.per-file-ignores]
+"__init__.py" = ["F401", "E402"] # imported but unused, Module level import not at top of file
+"qiskit/circuit/library/__init__.py" = ["F403"] # import * used
+"qiskit/extensions/__init__.py" = ["F403"]
+"qiskit/providers/fake_provider/__init__.py" = ["F403"]
+"qiskit/visualization/pulse/__init__.py" = ["F403"]
+"qiskit/opflow/operator_globals.py" = ["E741"] # Ambiguous variable name `I`
+"qiskit/quantum_info/operators/random.py" = ["F401"] # Should be fixed.
+"qiskit/quantum_info/random.py" = ["F401"] # Should be fixed.
+"qiskit/synthesis/discrete_basis/solovay_kitaev.py" = ["F821"] # Undefined name
+"qiskit/test/decorators.py" = ["F401"] # `qiskit.providers.aer` imported but unused
+"qiskit/tools/visualization.py"  = ["F403"]
+"qiskit/transpiler/passes/calibration/builders.py" = ["F401"]
+"qiskit/transpiler/passes/optimization/hoare_opt.py" = ["F401"]
+"qiskit/transpiler/passes/scheduling/calibration_creators.py" = ["F401"]
+"qiskit/transpiler/passes/scheduling/rzx_templates.py" = ["F401"]
+
+
```

Comment:
Per the ruff docs and some of your comments on docstrings in #1179 I'm wondering if we want to add

```yaml
[tool.ruff.pydocstyle]
convention = "google"
```

here too

https://github.com/charliermarsh/ruff#does-ruff-support-numpy--or-google-style-docstrings

Reply:
That's certainly worth trying.

Reply:
I added some doc string checks. But in order to make this more manageable, I plan to instead defer them to a later PR (provided we actually make a switch to ruff) However, several doc string errors/suggestions made by ruff have been merged in recent PRs.